grafanaplane/oncall/schedule.libsonnet

```diff
@@ -44,20 +44,34 @@ local forProvider = schedule.spec.parameters.forProvider;
       |||,
       [
         d.argument.new('name', d.T.string),
-        d.argument.new('shifts', d.T.object, default='{}'),
+        d.argument.new('namespace', d.T.string),
+        d.argument.new('shifts', d.T.object),
       ]
     ),
-    new(name, shifts={}): {
+    new(name, namespace, shifts): {
+      local this = self,
       scheduleName:: xtd.ascii.stringToRFC1123(name),
+      scheduleNamespace:: namespace,
+      scheduleShiftLabels:: {
+        // Shifts need to be looked up using a selector because the Schedule
+        // refers not to the claims, but to realized compositions with
+        // generated names. The Schedule cannot use a list of selectors,
+        // though, only one, so we inject this set of labels to the given
+        // shifts, and select with these.
+        'oncall.grafana.crossplane.io/schedule-claim-name': this.scheduleName,
+        'oncall.grafana.crossplane.io/schedule-claim-namespace': this.scheduleNamespace,
+      },
       schedule:
         schedule.new(self.scheduleName)
         + forProvider.withName(name)
         + forProvider.withType('calendar')
-        + forProvider.withShiftsRef([
-          forProvider.shiftsRef.withName(shift.metadata.name)
-          for shift in self.shifts
-        ]),
-      shifts: shifts,
+        + forProvider.shiftsSelector.withMatchLabels(this.scheduleShiftLabels),
+      shifts: [
+        // Inject matching labels to identify Shifts as belonging to this Schedule.
+        raw.oncall.v1alpha1.onCallShift.metadata.withLabels(this.scheduleShiftLabels)
```

Comment:
This won't work because the latter sets the labels on the claim resource while `selector.matchLabels` tells Crossplane to look at the managed resources.

I've got a theoretical solution in the pipeline: https://github.com/grafana/grafana-crossplane-libsonnet/pull/36/commits/88ae384dc389a8bf0a83fbec183de91f4caa662d

The idea is to set a label on the managed resource this way, then let `selector.matchLabels` find it, however we'll need to verify that the code in Crossplane can actually deal with a selector that returns multiple resources.